### PR TITLE
drivers: usb: device: Migrate to new logger

### PR DESCRIPTION
**Pull request by ydamigos**
Move to new logger subsystem.

Signed-off-by: Yannis Damigos <giannis.damigos@gmail.com>

### CODE REVIEW

**Review by loicpoulain — DISMISSED** [review]

**Comment by ydamigos** [comment]
Updated to resolve conflicts.

**Review by jfischer-no — DISMISSED** [review]

**Review by erwango — DISMISSED** [review]

**Review by galak — DISMISSED** [review]
Can you fix this for the change in LOG_MODULE_REGISTER to take the name as a param, and remove LOG_MODULE_NAME

**Comment by ydamigos** [comment]
@galak Done

**Comment by ydamigos** [comment]
recheck

**Review by finikorg — DISMISSED** [review]

**Comment by ydamigos** [comment]
Updated to migrate all usb device drivers to new logger.

**Comment by ydamigos** [comment]
@finikorg @jfischer-phytec-iot @galak @loicpoulain @erwango Could you check it again?

File: drivers/usb/device/usb_dc_dw.c [hunk 1]
```diff
@@ -587,7 +588,7 @@ static void usb_dw_handle_enum_done(void)
 	speed = (USB_DW->dsts & ~USB_DW_DSTS_ENUM_SPD_MASK) >>
 	    USB_DW_DSTS_ENUM_SPD_OFFSET;
 
-	SYS_LOG_DBG("USB ENUM DONE event, %s speed detected",
+	LOG_DBG("USB ENUM DONE event, %s speed detected",
 	    speed == USB_DW_DSTS_ENUM_LS ? "Low" : "Full");
```
Comment (finikorg) [inline drivers/usb/device/usb_dc_dw.c]: all multi-line logs need to be adjusted

File: drivers/usb/device/usb_dc_kinetis.c [hunk 2]
```diff
@@ -183,7 +184,7 @@ static int kinetis_usb_init(void)
 
 	irq_enable(CONFIG_USBD_KINETIS_IRQ);
 
-	SYS_LOG_DBG("");
+	LOG_DBG("");
```
Comment (finikorg) [inline drivers/usb/device/usb_dc_kinetis.c]: I assume this does not print function name, right? I think we need to invent some macro like networking and other subsys are doing to overcome missing function name limitation.

**Review by finikorg — CHANGES_REQUESTED** [review]
Main issues are: multi-line adjustments and missing function name. Look what I have done here: 9d5c8402fec1faf3cb2a85ca8aef7efe0afed319

File: drivers/usb/device/usb_dc_dw.c (hunk 1 above)
Comment (ydamigos) [reply drivers/usb/device/usb_dc_dw.c]: done

File: drivers/usb/device/usb_dc_kinetis.c (hunk 2 above)
Comment (ydamigos) [reply drivers/usb/device/usb_dc_kinetis.c]: Thanks for the pointer. Done.

File: drivers/usb/device/usb_dc_dw.c [hunk 3]
```diff
@@ -653,7 +645,7 @@ static void usb_dw_isr_handler(void)
 			xfer_size = (grxstsp & USB_DW_GRXSTSR_PKT_CNT_MASK) >>
 			    USB_DW_GRXSTSR_PKT_CNT_OFFSET;
 
-			SYS_LOG_DBG("USB OUT EP%d: RX_FLVL status %d, size %d",
+			LOG_DBG("USB OUT EP%d: RX_FLVL status %d, size %d",
 				ep_idx, status, xfer_size);
```
Comment (finikorg) [inline drivers/usb/device/usb_dc_dw.c]: Does this line need adjustment?

**Review by finikorg — COMMENTED** [review]
What about defining
USB_DBG()
USB_WRN()
USB_ERR()
the same way networking subsys does. See faf3068f6da9b1fb606cb47c2cb8530134c6ead4.
Better to use the same size for macros for easy adjustment of multi-line logs (i.e. LOG_DBG => USB_DBG)

File: drivers/usb/device/usb_dc_dw.c (hunk 3 above)
Comment (finikorg) [reply drivers/usb/device/usb_dc_dw.c]: It also makes sense to add function name to the logs.

**Comment by jukkar** [comment]
Just fyi and not really relevant to this PR, but I have USB NET conversion to new logging at https://github.com/zephyrproject-rtos/zephyr/pull/8816/commits/4a89de21712fb8f4524b55072ba11c0be8d898a5

File: drivers/usb/device/usb_dc_dw.c (hunk 3 above)
Comment (finikorg) [reply drivers/usb/device/usb_dc_dw.c]: Or, maybe, conversion LOG_DBG => USB_DBG might be done in the follow up patch, just fix adjustment.

Comment (ydamigos) [reply drivers/usb/device/usb_dc_dw.c]: I added USB_* macros.

**Review by finikorg — APPROVED** [review]
Thanks, looks good, USB_* macros might be put later to some common header.

File: samples/subsys/usb/cdc_acm/prj.conf [hunk 4]
```diff
@@ -4,7 +4,7 @@ CONFIG_GPIO=y
 CONFIG_USB=y
 CONFIG_USB_DEVICE_STACK=y
 CONFIG_USB_DEVICE_PRODUCT="Zephyr CDC ACM sample"
-CONFIG_SYS_LOG_USB_DRIVER_LEVEL=1
+CONFIG_USB_DRIVER_LOG_LEVEL=1
```
Comment (jfischer-phytec-iot) [inline samples/subsys/usb/cdc_acm/prj.conf]: Please add CONFIG_LOG=y as well to all prj*.conf

**Review by jfischer-no — CHANGES_REQUESTED** [review]

**Review by loicpoulain — APPROVED** [review]

File: samples/subsys/usb/cdc_acm/prj.conf (hunk 4 above)
Comment (ydamigos) [reply samples/subsys/usb/cdc_acm/prj.conf]: Done

**Review by jfischer-no — APPROVED** [review]

**Comment by ydamigos** [comment]
Shippable is not happy because `samples/net/zperf` tests fail on quark_se_c1000_devboard due to RAM overflow.
@finikorg I don't like to disable tests on quark_se_c1000_devboard for `samples/net/zperf`, is there another way to make shippable happy?

**Comment by finikorg** [comment]
@ydamigos Fixed.

BTW: @ydamigos @jfischer-phytec-iot @loicpoulain @carlescufi Is somebody is working already on converting USB Device stack to the new logger? I can do this but do not like to do double work.

**Comment by loicpoulain** [comment]
> Is somebody is working already on converting USB Device stack to the new logger? 

@ydamigos, not on my side.

**Comment by ydamigos** [comment]
> Is somebody is working already on converting USB Device stack to the new logger? I can do this but do not like to do double work.

@findlayfeng Thanks. I already created PR #10168.

**Comment by ydamigos** [comment]
Updated to resolve conflicts.

File: drivers/usb/device/usb_dc_kinetis.c [hunk 5]
```diff
@@ -183,7 +189,7 @@ static int kinetis_usb_init(void)
 
 	irq_enable(CONFIG_USBD_KINETIS_IRQ);
 
-	SYS_LOG_DBG("");
+	USB_DBG("");
```
Comment (nashif) [inline drivers/usb/device/usb_dc_kinetis.c]: do we really need those? why?

File: drivers/usb/device/usb_dc_kinetis.c [hunk 6]
```diff
@@ -216,15 +222,15 @@ int usb_dc_reset(void)
 		       USB_INTEN_STALLEN_MASK |
 		       USB_INTEN_ERROREN_MASK);
 
-	SYS_LOG_DBG("");
+	USB_DBG("");
```
Comment (nashif) [inline drivers/usb/device/usb_dc_kinetis.c]: do we really need those? why?

File: drivers/usb/device/usb_dc_kinetis.c [hunk 7]
```diff
@@ -246,7 +252,7 @@ int usb_dc_attach(void)
 
 int usb_dc_detach(void)
 {
-	SYS_LOG_DBG("");
+	USB_DBG("");
```
Comment (nashif) [inline drivers/usb/device/usb_dc_kinetis.c]: ditto

File: drivers/usb/device/usb_dc_dw.c [hunk 8]
```diff
@@ -25,8 +25,15 @@
 #include "clk.h"
 #endif
 
-#define SYS_LOG_LEVEL CONFIG_SYS_LOG_USB_DRIVER_LEVEL
-#include <logging/sys_log.h>
+#define LOG_LEVEL CONFIG_USB_DRIVER_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(usb_dc_dw);
+
+#define USB_DBG(fmt, ...) LOG_DBG("(%p): %s: " fmt, k_current_get(), \
```
Comment (nashif) [inline drivers/usb/device/usb_dc_dw.c]: why are we redefining this here? is it to get the function/line no only? We should avoid this deviation and introduction of subsystem specific macros and instead look how we can get this functionality from the core logger. @nordic-krch, is there a way for us to include __func__ in the output in a generic way?

**Review by nashif — CHANGES_REQUESTED** [review]

**Comment by ydamigos** [comment]
@nashif All your comments are answered in PR #10168. I am closing this PR in favor of #10168.

**Pull request closed without merging**